Block/BlockContextManager.php

```diff
@@ -214,4 +212,31 @@ protected function setDefaultExtraCacheKeys(BlockContextInterface $blockContext,
             $blockContext->setSetting('extra_cache_keys', $extraCacheKeys);
         }
     }
+
+    private function resolve(BlockInterface $block, $settings)
+    {
+        $optionsResolver = new OptionsResolver();
+
+        $this->configureSettings($optionsResolver, $block);
+
+        $service = $this->blockService->get($block);
+        $service->setDefaultSettings($optionsResolver, $block);
+
+        $reflector = new \ReflectionMethod($service, 'setDefaultSettings');
```

Comment:
Something like this.

```
$this->reflectionCache = [
 'class\path' => ['isOldOverwritten' => true|false, 'isNewOverwritten' => true|false
];


$class = get_class($service);
if ($this->reflectionCache[$class]['isOldOverwritten'] && !$this->reflectionCache[$class]['isNewOverwritten']) {
        trigger_error('The Sonata\BlockBundle\Block\BlockServiceInterface::setDefaultSettings() method is deprecated since version 2.3 and will be removed in 3.0. Use configureSettings() instead. This method will be added to the BlockServiceInterface with SonataBlockBundle 3.0.', E_USER_DEPRECATED);
 }
```